_pytest/hookspec.py

```diff
@@ -156,6 +156,12 @@ def pytest_pyfunc_call(pyfuncitem):
 def pytest_generate_tests(metafunc):
     """ generate (multiple) parametrized calls to a test function."""
 
+@hookspec(firstresult=True)
+def pytest_make_parametrize_id(val):
```

Comment:
I thought it would be useful to also receive the `config` object. Some plugin might add some custom options that might affect this.